### PR TITLE
Azure kusto service upgrade sdk to version 2022-02-01

### PR DESCRIPTION
**Pull request by liuwuliuyun**
Upgrade kusto sdk version to latest and prepare for adding new properties.

### CODE REVIEW

**Comment by liuwuliuyun** [comment]
MicrosoftTeams-image
Regression test evidence.

**Comment by liuwuliuyun** [comment]
FYI, the vnet test fail is not due to the version update, it is an exsiting bug on our main branch before updating.

**Comment by ms-henglu** [comment]
image

There's a failed test, but it's a temporary failure, no need to worry.

**Comment by liuwuliuyun** [comment]
Rerun above and pass
image

**Comment by liuwuliuyun** [comment]
> We still have a test failure here:
> 
> ```
> ------- Stdout: -------
> === RUN   TestAccKustoCluster_zones
> === PAUSE TestAccKustoCluster_zones
> === CONT  TestAccKustoCluster_zones
>     testcase.go:110: Step 1/1 error: After applying this test step and performing a `terraform refresh`, the plan was not empty.
>         stdout
>         
>         
>         Terraform used the selected providers to generate the following execution
>         plan. Resource actions are indicated with the following symbols:
>         -/+ destroy and then create replacement
>         
>         Terraform will perform the following actions:
>         
>           # azurerm_kusto_cluster.test must be replaced
>         -/+ resource "azurerm_kusto_cluster" "test" {
>               ~ data_ingestion_uri            = "https://ingest-acctestkchfjj2.eastus2.kusto.windows.net/" -> (known after apply)
>               - double_encryption_enabled     = false -> null
>               ~ id                            = "/subscriptions/*******/resourceGroups/acctestRG-220608200522335526/providers/Microsoft.Kusto/Clusters/acctestkchfjj2" -> (known after apply)
>               - language_extensions           = [] -> null
>                 name                          = "acctestkchfjj2"
>               - tags                          = {} -> null
>               ~ trusted_external_tenants      = [] -> (known after apply)
>               ~ uri                           = "https://acctestkchfjj2.eastus2.kusto.windows.net/" -> (known after apply)
>               ~ zones                         = [ # forces replacement
>                   - "1",
>                   + "3",
>                 ]
>                 # (8 unchanged attributes hidden)
>         
>                 # (1 unchanged block hidden)
>             }
>         
>         Plan: 1 to add, 0 to change, 1 to destroy.
> --- FAIL: TestAccKustoCluster_zones (1266.27s)
> FAIL
> ```

I have checked this one, this is related to kusto service API behavior and I will reach out to them to see what is the expected behavior of this API and determine whether this is suitable for us to add this "zone" property to azure provider of terraform.

**Comment by liuwuliuyun** [comment]
The API behavior is unstable from my test case, so I will mark this PR as draft for now until I heard from kusto service team.

**Comment by liuwuliuyun** [comment]
Just got confirmation from kusto service team that they are aware of this bug and will fix in 1~2 weeks. When they perform the fix,  the failed test case will pass automatically. As a result, I reopen the MR for review.

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @liuwuliuyun 

Thanks for pushing those changes - apologies for the delayed re-review here.

Taking a look through this it appears there's one thing to fix, but if we can fix that up then this should otherwise be good to go 👍 

Thanks!

File: internal/services/kusto/kusto_cluster_resource_test.go [hunk 1]
```diff
@@ -806,7 +806,11 @@ resource "azurerm_kusto_cluster" "test" {
 func (KustoClusterResource) vnet(data acceptance.TestData) string {
 	return fmt.Sprintf(`
 provider "azurerm" {
-  features {}
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
```
[the remark this answers is not in the document]
Comment (liuwuliuyun) [reply internal/services/kusto/kusto_cluster_resource_test.go:812]: Oh, this is not needed now, to keep our commit history clean, I raise a new PR https://github.com/hashicorp/terraform-provider-azurerm/pull/17457. Actually, the service team is making a release now and once their service is released, I will provide clean test results on that new PR.

**Comment by liuwuliuyun** [comment]
Close this PR to keep up with latest upgrade with new PR after kusto service team complete release.

**Pull request closed without merging**